engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -1516,6 +1531,20 @@ public void run()
     }
   }
 
+  /*
+   * Generates groupId for operator error event.
+   * The events generated by downstream operators redeployment should use this groupId
+   */
+  private long getUndeployGroupId(int[] operators)
```

Comment:
Right, when we started that was the thought process, and I actually have tried coding that way. The limitations I face:
1. When we reportError, "StreamingContainerParent" will have to start tracking ContainerDeployRequest.
2. In parallel the heartbeats will keep coming and we could send operator stop event in it.

Doing it, I found there are more places where I have to create and keep tracking the new data structures. Instead if I do it the way right now, it goes in the current flow which is already there. It is complicated but at least it's not new.